### PR TITLE
Feat (nn/sdpa): quantization of scaled dot-product attention

### PR DESCRIPTION
**Pull request by nickfraser**
## Reason for this PR

Make it easier for users to quantize attention layers.

## Changes Made in this PR

Achieved by providing:
 - [x] A modular equivalent to the `torch.nn.functional.scaled_dot_product_attention` functional
 - [x] A quantized version of this module
 - [x] Adding code to convert the three options

## Testing Summary

Tests:
 - [x] Layer replacement test in LLM entry-point
 - [x] Basic accuracy test for OPT
 - [ ] ~Basic graph replacement test~ (covered by LLM entry-point test)
 - [x] SDPA & Quant SDPA forward tests

## Risk Highlight

Adapted from pseudocode in PyTorch's documentation. Otherwise, this code barely touches any existing code, that shouldn't break any existing Brevitas features.

- [x] This PR includes code from another work (please detail).
- [ ] This PR contains API-breaking changes.
- [ ] This PR depends on work in another PR (please provide links/details).
- [ ] This PR introduces new dependencies (please detail).
- [x] There are coverage gaps not covered by tests.
- [ ] Documentation updates required in subsequent PR.

## Checklist

- [ ] Code comments added to any hard-to-understand areas, if applicable.
- [ ] Changes generate no new warnings.
- [ ] Updated any relevant tests, if applicable.
- [ ] No conflicts with destination `dev` branch.
- [ ] I reviewed my own code changes.
- [ ] Initial CI/CD passing.
- [ ] 1+ reviews given, and any review issues addressed and approved.
- [ ] Post-review full CI/CD passing.

### CODE REVIEW

**Comment by nickfraser** [comment]
We should merge #1088 before this.

File: src/brevitas/nn/quant_sdpa.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            k_transposed_quant=Int8ActPerTensorFloat,
+            v_quant=Int8ActPerTensorFloat,
+            attn_output_quant=None,
+            **kwargs) -> None:
+        super(QuantScaledDotProductAttention, self).__init__()
+
+        def filter_kwargs(prefix):
+            return {k[len(prefix):]: v for k, v in kwargs.items() if k.startswith(prefix)}
+
+        self.q_scaled_quant = QuantIdentity(act_quant=q_scaled_quant, **filter_kwargs('q_scaled_'))
```
Comment (Giuseppe5) [inline src/brevitas/nn/quant_sdpa.py:131]: This should work, but I believe we also have a QuantLayer argument to specify the prefix, which should be equivalent. Rarely used so it's worth double checking.

File: src/brevitas_examples/llm/main.py [hunk 2]
```diff
@@ -593,6 +604,10 @@ def parse_args(args):
         type=float,
         default=1e-4,
         help='Minimum value to clamp scale to when using bf16 or fp16 quantization.')
+    parser.add_argument(
```
Comment (Giuseppe5) [inline src/brevitas_examples/llm/main.py:607]: Regenerate README

Comment (nickfraser) [reply src/brevitas_examples/llm/main.py:607]: Done

File: src/brevitas/nn/quant_sdpa.py (hunk 1 above)
Comment (nickfraser) [reply src/brevitas/nn/quant_sdpa.py:131]: Added test for this.

**Comment by nickfraser** [comment]
Note, when `--quant-sdpa` is applied, the matrix multiplies (i.e., Q,K,V & attention softmax) are quantized in the same way as inputs to linear layers. This is slightly different to the current version of the stable diffusion example, which can have different formats for attention quantization and linear/conv layer quantization.

**Review by Giuseppe5 — APPROVED** [review]
I love it
